src/main/java/com/softserve/teamproject/controller/GroupController.java

```diff
@@ -102,13 +102,26 @@ public void deleteGroup(@PathVariable Integer id, Principal principal) {
    * @param principal is an authenticated user
    */
   @PutMapping(value = "/groups/{id}")
-  public void editGroup(@RequestBody @Valid Group group, @PathVariable Integer id, Principal principal) {
+  public GroupResource editGroup(@RequestBody Group group, @PathVariable Integer id, Principal principal) {
     group.setId(id);
     Status currentStatus = groupService.getGroupById(id).getStatus();
-    groupService.updateGroup(group, currentStatus, principal.getName());
+    groupService.fieldsCheck(group);
+    return groupService.updateGroup(group, currentStatus, principal.getName());
   }
 
-  /* Get groups by filter
+  /**
+   * Gets group by id.
+   *
+   * @param id is id of a group we want to find.
+   * @return group with entered id.
+   */
+  @GetMapping(value = "/groups/{id}")
+  public Group getGroupById(@PathVariable Integer id) {
```

Comment:
GroupResource returns Group in HATEOAS style, the fact this works, doesn't mean it corresponds to our requirements